src/material/core/tokens/_token-utils.scss

```diff
@@ -0,0 +1,77 @@
+@use 'sass:map';
+@use '@material/theme/custom-properties' as mdc-custom-properties;
+@use '@material/theme/theme' as mdc-theme;
+@use '../mdc-helpers/mdc-helpers';
+@use '../theming/palette';
+
+// Placeholder color config that can be passed to token getter functions when generating token
+// slots.
+$placeholder-color-config: (
+    primary: palette.$red-palette,
+    accent: palette.$red-palette,
+    warn: palette.$red-palette,
+    is-dark: false,
+    foreground: palette.$light-theme-foreground-palette,
+    background: palette.$light-theme-background-palette,
+);
+
+$_placeholder-typography-level-config: mdc-helpers.typography-config-level-from-mdc(headline1);
```

Comment:
nit: maybe we should use one of the body sizes? This will cause some huge text if the theme isn't loaded.

Reply:
These are purely placeholder values. They're used to create a token slots map where the tokens have valid non-null values which can then be used to generate the slots, e.g.: `font-size: var(--mdc-list-list-item-font-size)`, but the actual value is not used.